lib/deep-equal.js

```diff
@@ -105,7 +108,8 @@ function deepEqualCyclic(first, second) {
             if (
                 type1 !== type2 ||
                 class1 !== class2 ||
-                keys1.length !== keys2.length
+                keys1.length !== keys2.length ||
+                (!!name1 && name2 && name1 !== name2)
```

Comment:
When used in a boolean expression, you don't need to use `!!` to coerce a truthy value to a boolean, as the JavaScript engine will do that for you.

Reply:
So true :)